### DISCUSSION TITLE
General Guidance for Migrating from 1.3.1 to 3.0.2

### DISCUSSION THREAD

**Discussion by julianhall** [spring-projects/spring-net#256] [General]
Hi, 

We currently have a spring.net web application that's been deployed for quite some time on 1.3.1/asp.net 48.
We would like to modernize & upgrade to .net8 core. I can't seem to find any upgrade/migration documentation to assist with it. 
I understand that:

- all dependencies will need to be upgraded
- and migrating web applications from .net framework to .net core is not straight forward and there are many guides out there.

I'm wondering if anyone has done this here from a Spring.NET web app?
I assume the Spring.Web module in the 3.0.2 release is not to be used anymore as it still targets framework 4.8? 
We seem to only use the following from the 1.3.1 Spring.Web module:

```
- Spring.Web.Support.MappingHandlerFactoryConfigurer
- Spring.Context.Support.HttpApplicationConfigurer
```

Not sure how I would move from these.  Any assistance would be great! 

Thanks - Jules

**Comment by julianhall** [comment]
hmm... crickets.   
Willing to pay for any consultation/work.

**Reply by lahma** [reply]
I think the problem is that there's no "general" way to do this. It all depends on how the current migration target is structured, which technologies it's using and what are the future targets. If you are aiming to use ASP.NET Core, then the default MS DI might be easier choice. 

Spring.NET is basically in maintenance mode without new feature development (like ASP.NET Core integration).

**Reply by julianhall** [reply]
Thanks @lahma - I suspected as much, but wanted to confirm.

Essentially you are saying if we are going to spend the time to try and upgrade to .net8 / ASP.NET Core, then we should look at migrating from Spring.NET at the same time and leverage MS DI? 

Thanks - Jules

**Reply by lahma** [reply]
Depends.. You could probably also bridge to Spring's container if it would otherwise be too much to migrate at one go. Spring.NET doesn't have proper scoped lifecycle so you would have to rely on singletons and transient objects.

**Reply by julianhall** [reply]
Thanks for the reply @lahma, that is our current dilemma - bridge or full convert.... appreciate the input.
